components/ShoppingItem.js

```diff
@@ -1,24 +1,57 @@
 import styled from "styled-components";
 import Link from "next/link";
+import { useState } from "react";
+import Delete from "@/public/icons/delete.svg";
+
+export default function ShoppingItem({
+  id,
+  quantity,
+  name,
+  category,
+  onDeleteItem,
+}) {
+  const [isDeleteConfirmVisible, setIsDeleteConfirmVisible] = useState(false); // Zustand für den Bestätigungsdialog
+
+  // Funktion zum Anzeigen des Bestätigungsdialogs
+  function handleShowConfirm() {
+    setIsDeleteConfirmVisible(true);
+  }
+
+  // Funktion zum Verstecken des Bestätigungsdialogs
+  function handleHideConfirm() {
+    setIsDeleteConfirmVisible(false);
+  }
 
-export default function ShoppingItem({ id, quantity, name, category }) {
   return (
     <ItemContainer $category={category}>
-      <ItemQuantity>{quantity}</ItemQuantity>
-      <ItemName>{name}</ItemName>
-      <ItemCategory>{category}</ItemCategory>
-      <StyledLink href={`/${id}`}>Details</StyledLink>
+      {isDeleteConfirmVisible ? (
+        <>
+          <ConfirmDialog>Delete item?</ConfirmDialog>
+          <ConfirmButton onClick={() => onDeleteItem(id)}>Yes</ConfirmButton>
+          <CancelButton onClick={handleHideConfirm}>No</CancelButton>
+        </>
+      ) : (
+        <>
+          <ItemQuantity>{quantity}</ItemQuantity>
+          <ItemName>{name}</ItemName>
+          <ItemCategory>{category}</ItemCategory>
+          <StyledLink href={`/${id}`}>Details</StyledLink>
+          <StyledButton onClick={handleShowConfirm}>
```

Comment:
If you combine the functions above into a toggle function, you would of course have to update this.